hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java

```diff
@@ -4403,4 +4403,11 @@ public void xmltable_sqlserver() {
 	public void xmltable_sybasease() {
 		functionRegistry.register( "xmltable", new SybaseASEXmlTableFunction( typeConfiguration ) );
 	}
+
+	/**
+	 * Oracle extract() function
+	 */
+	public void extract_oracle(Dialect dialect) {
+		functionRegistry.register( "extract", new OracleExtractFunction( dialect, typeConfiguration ) );
```

Comment:
Let's just instantiate the custom function directly in `OracleDialect`, no need to add a function here.